File: API/Models/ProjectTasks.cs


namespace API.Models
{
    public class ProjectTasks
    {
        // Primary Key
        public int ProjectTasksID { get; set; }

        // Properties
        public string ProjectTasksName { get; set; } = String.Empty;
        public string ProjectTasksDescription { get; set; } = String.Empty;
        public string ProjectTasksStatus { get; set; } = String.Empty;
        public string ProjectTasksStartDate { get; set; } = String.Empty;
        public string ProjectTasksEndDate { get; set; } = String.Empty;

        // Enumaration
        public ProjectTaskStatusEnum ProjectTaskStatusEnumC { get; set; } 


        // Foreign Keys 1 to Many
        public List<AppUser> AppUsers { get; set; } = new List<AppUser>();

        // Foreign Keys 1 to 1
        // Project Navigation
        public int? ProjectID { get; set; }
        public string? ProjectName { get; set; } = String.Empty;
        public Project? Project { get; set; }

        // Phase Navigation
        public int? ProjectPhaseID { get; set; }
        public string? ProjectPhaseName { get; set; } = String.Empty;
        public ProjectPhase? ProjectPhase { get; set; }
    }
}

File: FRONT/src/app/models/projectTasks.ts
import { ProjectPhase } from "./projectPhase";

export interface ProjectTasks {
    projectTasksID: number;
    projectTasksName: string;
    projectTasksDescription: string;
    projectTasksStartDate: string;
    projectTasksEndDate: string;
    projectTasksStatus: string;
    projectID: number;
    projectPhase: Array<ProjectPhase>;
    projectPhaseID: number;
}